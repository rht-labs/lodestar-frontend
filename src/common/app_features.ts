// All roles defined by the backend to be used in the frontend should be enumerated in this object.

export const APP_FEATURES = {
  admin: 'admin',
  writer: 'writer',
  reader: 'reader',
  notifications: 'notifications',
  engagementCardIcons: 'engagementCardIcons',
  resetUser: 'resetUser',
};

export type AppFeature = keyof typeof APP_FEATURES;
