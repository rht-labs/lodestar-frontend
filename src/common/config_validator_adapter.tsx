import { ValidatorFactory } from '../schemas/validators';
import { EngagementFormConfig } from '../schemas/engagement_config';

export const getValidatorsFromFormOptions = (
  formOptions: EngagementFormConfig = {}
) =>
  Object.keys(formOptions || {}).reduce(
    (previousSchemaGroups, currentGroupKey) => {
      return {
        ...previousSchemaGroups,
        ...Object.keys(formOptions[currentGroupKey] ?? {}).reduce(
          (previousGroupFields, currentFieldKey) => ({
            ...previousGroupFields,
            [currentFieldKey]: (
              formOptions?.[currentGroupKey]?.[currentFieldKey]?.validators ||
              []
            ).map(ValidatorFactory),
          }),
          {}
        ),
      };
    },
    {}
  );
