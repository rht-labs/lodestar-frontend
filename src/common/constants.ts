export const APPLICATION_CONTENT_CONTAINER_ID = 'main-application-content';
