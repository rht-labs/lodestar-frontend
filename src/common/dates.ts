import { format as formatDate, utcToZonedTime } from 'date-fns-tz';
export const formatUtcDate = (date?: Date): string | undefined => {
  if (!date) {
    return '';
  }
  const dateString = formatDate(utcToZonedTime(date, 'UTC'), 'MMM dd, yyyy', {
    timeZone: 'UTC',
  });
  return dateString;
};
