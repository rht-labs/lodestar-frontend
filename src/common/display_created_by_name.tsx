import React from 'react';

export function DisplayCreatedByName({
  userFromServer,
  lastUpdatedBy,
}: {
  userFromServer?: string;
  lastUpdatedBy?: string;
}) {
  return <>{!!userFromServer ? userFromServer : lastUpdatedBy}</>;
}
