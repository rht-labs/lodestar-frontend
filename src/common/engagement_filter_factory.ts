import {
  EngagementFilter,
  EngagementSortFields,
} from '../schemas/engagement_filter';
import { Engagement, getEngagementStatus } from '../schemas/engagement';

function hasSearchTermMatch(engagement: Engagement, searchTerm: string = '') {
  if (!searchTerm || !engagement) {
    return true;
  }
  const cleanedSearchTerm = searchTerm?.trim()?.toLowerCase();
  const cleanedCustomerName = engagement.customer_name.toLowerCase().trim();
  const cleanedProjectName = engagement.project_name.toLowerCase().trim();
  return (
    cleanedProjectName.includes(cleanedSearchTerm) ||
    cleanedCustomerName.includes(cleanedSearchTerm)
  );
}

/**
 * The Engagement Filter Factory accepts a filter definition as a parameter.
 * It parses the filter definition and returns a single filter function that
 * filters an array of Engagements based on the terms of the filter definition.
 */
export const engagementFilterFactory = (
  filter?: EngagementFilter
): ((engagement: Engagement) => boolean) => {
  if (!filter || Object.keys(filter).length === 0) {
    return () => true;
  }
  return (engagement: Engagement) => {
    const filterTestResults: boolean[] = [];
    if (filter.allowedStatuses) {
      filterTestResults.push(
        filter.allowedStatuses.includes(getEngagementStatus(engagement))
      );
    }
    if (filter.searchTerm) {
      filterTestResults.push(
        hasSearchTermMatch(engagement, filter?.searchTerm)
      );
    }
    /**
     * If filterTestResult has a length of 0, this means no tests were performed.
     * The default behavior of the filter should be to show engagements.
     * Therefore, we return true
     */
    if (filterTestResults.length === 0) {
      return true;
    }
    /**
     * If filterTestResults has results, we only return if every result is true.
     */
    return filterTestResults.every(r => !!r);
  };
};

const engagementSorters = (sortOrderMultiplier: number = 1) => {
  return {
    startDate: (a: Engagement, b: Engagement) =>
      (a.start_date > b.start_date ? 1 : -1) * sortOrderMultiplier,

    endDate: (a: Engagement, b: Engagement) =>
      (a.end_date > b.end_date ? 1 : -1) * sortOrderMultiplier,

    customerName: (a: Engagement, b: Engagement) =>
      (a.customer_name > b.customer_name ? 1 : -1) * sortOrderMultiplier,

    projectName: (a: Engagement, b: Engagement) =>
      (a.project_name > b.project_name ? 1 : -1) * sortOrderMultiplier,
  };
};

/**
 * The Engagement Sort Factory parses the `sort` field of the Engagement filter definition.
 * It returns a single function that can be passed to the `sort` method of an Engagement array.
 */
export const engagementSortFactory = ({ sort }: EngagementFilter = {}): ((
  a: Engagement,
  b: Engagement
) => number) => {
  /**
   * The sort order multiplier translates the isAscending boolean field into
   * an integer that can invert the sort order of the array.
   */
  const sortOrderMultiplier =
    sort?.isAscending === undefined || sort?.isAscending ? 1 : -1;
  if (sort?.sortField === EngagementSortFields.startDate) {
    return engagementSorters(sortOrderMultiplier).startDate;
  }
  if (sort?.sortField === EngagementSortFields.endDate) {
    return engagementSorters(sortOrderMultiplier).endDate;
  }
  if (sort?.sortField === EngagementSortFields.customerName) {
    return engagementSorters(sortOrderMultiplier).customerName;
  }
  if (sort?.sortField === EngagementSortFields.projectName) {
    return engagementSorters(sortOrderMultiplier).projectName;
  }
  return () => 0;
};
