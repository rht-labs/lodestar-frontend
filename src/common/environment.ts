export class Environment {
  public getValue(key: string) {
    return process.env[key];
  }
}
