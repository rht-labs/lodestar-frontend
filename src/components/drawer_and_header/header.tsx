import React from 'react';
import { UserDropdown } from './user_dropdown';
import { Link, useLocation } from 'react-router-dom';
import { Notification } from './notification';
import {
  Brand,
  Nav,
  NavItem,
  NavList,
  PageHeader,
  PageHeaderTools,
  PageHeaderToolsGroup,
  PageHeaderToolsItem,
  TooltipPosition,
} from '@patternfly/react-core';
import { Tooltip } from '@patternfly/react-core';

import { HelpIcon, LightbulbIcon } from '@patternfly/react-icons';

export interface HeaderProps {
  isDrawerOpen: boolean;
  onNavToggle: () => void;
  onNotificationClick: () => void;
}

export function Header(props: HeaderProps) {
  const iconPad: React.CSSProperties = {
    paddingBottom: 0,
    paddingTop: 0,
  };
  const { pathname } = useLocation();
  return (
    <PageHeader
      showNavToggle
      isNavOpen={props.isDrawerOpen}
      onNavToggle={props.onNavToggle}
      logo={
        <Brand
          alt="Open Innovation Labs"
          src={`${process.env.PUBLIC_URL}/oil_logo.png`}
        />
      }
      headerTools={
        <PageHeaderTools>
          <PageHeaderToolsGroup>
            <PageHeaderToolsItem>
              <Nav theme="dark" variant="horizontal">
                <NavList>
                  <NavItem
                    id="aboutLink"
                    itemId={2}
                    isActive={pathname === '/about'}
                  >
                    <Link style={iconPad} to="/app/about" data-cy={'about_link'}>
                      <Tooltip
                        content={'About'}
                        entryDelay={0}
                        exitDelay={10}
                        position={TooltipPosition.bottom}>
                        <HelpIcon title="About" />
                      </Tooltip>
                    </Link>
                  </NavItem>
                  <NavItem
                    id="feedback"
                    itemId={3}
                    isActive={pathname === '/about'}
                  >
                    <Link
                      to="/app/requestfeature"
                      title="Feedback and Feature Request"
                    >
                      <Tooltip
                        content={'Feedback and Feature Request'}
                        entryDelay={0}
                        exitDelay={10}
                        position={TooltipPosition.bottom}>
                        <LightbulbIcon />
                      </Tooltip>
                    </Link>
                  </NavItem>
                  <NavItem id="notifications" itemId={4}>
                    <Tooltip
                      content={'Notifications'}
                      entryDelay={0}
                      exitDelay={10}
                      position={TooltipPosition.bottom}
                    >
                      <Notification
                        onNotificationClick={props.onNotificationClick}
                      />
                    </Tooltip>
                  </NavItem>
                </NavList>
              </Nav>
            </PageHeaderToolsItem>
            <PageHeaderToolsItem>
              <UserDropdown />
            </PageHeaderToolsItem>
          </PageHeaderToolsGroup>
        </PageHeaderTools>
      }
    />
  );
}
