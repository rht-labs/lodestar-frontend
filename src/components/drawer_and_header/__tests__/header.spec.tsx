import React from 'react';
import { MemoryRouter } from 'react-router';
import { act, render, waitForDomChange } from '@testing-library/react';
import { Header } from '../header';
import { TestStateWrapper } from '../../../common/test_state_wrapper';

describe('Notification UI ', () => {
  test('should match snapshot', async () => {
    await act(async () => {
      const rendered = render(
        <MemoryRouter>
          <TestStateWrapper>
            <Header
              onNotificationClick={() => {}}
              isDrawerOpen={true}
              onNavToggle={() => null}
            />
          </TestStateWrapper>
        </MemoryRouter>
      );
      await waitForDomChange();
      expect(rendered).toMatchSnapshot();
    });
  });
});
