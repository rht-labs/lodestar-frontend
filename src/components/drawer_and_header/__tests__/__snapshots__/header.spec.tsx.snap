// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification UI  should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="pf-c-page__header"
      >
        <div
          class="pf-c-page__header-brand"
        >
          <div
            class="pf-c-page__header-brand-toggle"
          >
            <button
              aria-controls="page-sidebar"
              aria-expanded="true"
              aria-label="Global navigation"
              class="pf-c-button pf-m-plain"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              id="nav-toggle"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                  transform=""
                />
              </svg>
            </button>
          </div>
          <a
            class="pf-c-page__header-brand-link"
          >
            <img
              alt="Open Innovation Labs"
              class="pf-c-brand"
              src="/oil_logo.png"
            />
          </a>
        </div>
        <div
          class="pf-c-page__header-tools"
        >
          <div
            class="pf-c-page__header-tools-group"
          >
            <div
              class="pf-c-page__header-tools-item"
            >
              <nav
                aria-label="Global"
                class="pf-c-nav pf-m-horizontal"
                data-ouia-component-id="2"
                data-ouia-component-type="PF4/Nav"
                data-ouia-safe="true"
              >
                <button
                  aria-label="Scroll left"
                  class="pf-c-nav__scroll-button"
                  disabled=""
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                      transform=""
                    />
                  </svg>
                </button>
                <ul
                  class="pf-c-nav__list"
                >
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      data-cy="about_link"
                      href="/app/about"
                      style="padding-bottom: 0px; padding-top: 0px;"
                    >
                      <svg
                        aria-labelledby="icon-title-0"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 64 1024 1024"
                        width="1em"
                      >
                        <title
                          id="icon-title-0"
                        >
                          About
                        </title>
                        <path
                          d="M512.059-73.143c-282.338 0-512.059 229.673-512.059 512.025 0 282.235 229.721 511.975 512.059 511.975 282.281 0 511.941-229.735 511.941-511.975 0.005-282.352-229.659-512.025-511.941-512.025zM512.059 826.523c-213.826 0-387.728-173.856-387.728-387.643 0-213.89 173.904-387.694 387.728-387.694 213.717 0 387.671 173.803 387.671 387.694 0.005 213.785-173.957 387.643-387.671 387.643zM585.143 164.571v109.714c0 4.951-1.808 9.237-5.429 12.857s-7.906 5.429-12.857 5.429h-109.714c-4.953 0-9.239-1.808-12.857-5.429s-5.429-7.906-5.429-12.857v-109.714c0-4.951 1.81-9.237 5.429-12.857s7.904-5.429 12.857-5.429h109.714c4.951 0 9.237 1.808 12.857 5.429s5.429 7.906 5.429 12.857zM521.616 365.714c118.343 0 214.286 88.965 214.286 187.429s-95.943 178.286-214.286 178.286c-173.045 0-208.091-93.714-213.963-171.113 0-7.936 6.729-9.838 15.872-9.838s108.073 0 113.143 0c6.096 0 14.475 0.633 17.554 10.571 0 48.857 135.968 54.953 135.968-7.904 0-31.506-29.717-63.817-68.571-66.194s-82.286-7.607-82.286-54.199c0-13.022 0-25.673 0-44.693 0-19.015 9.717-22.343 27.431-22.343s54.853-0.002 54.853-0.002z"
                          transform="rotate(180 0 512) scale(-1 1)"
                        />
                      </svg>
                    </a>
                  </li>
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      href="/app/requestfeature"
                      title="Feedback and Feature Request"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 352 512"
                        width="1em"
                      >
                        <path
                          d="M96.06 454.35c.01 6.29 1.87 12.45 5.36 17.69l17.09 25.69a31.99 31.99 0 0 0 26.64 14.28h61.71a31.99 31.99 0 0 0 26.64-14.28l17.09-25.69a31.989 31.989 0 0 0 5.36-17.69l.04-38.35H96.01l.05 38.35zM0 176c0 44.37 16.45 84.85 43.56 115.78 16.52 18.85 42.36 58.23 52.21 91.45.04.26.07.52.11.78h160.24c.04-.26.07-.51.11-.78 9.85-33.22 35.69-72.6 52.21-91.45C335.55 260.85 352 220.37 352 176 352 78.61 272.91-.3 175.45 0 73.44.31 0 82.97 0 176zm176-80c-44.11 0-80 35.89-80 80 0 8.84-7.16 16-16 16s-16-7.16-16-16c0-61.76 50.24-112 112-112 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
                          transform=""
                        />
                      </svg>
                    </a>
                  </li>
                  <li
                    class="pf-c-nav__item"
                  >
                    <button
                      aria-label="Notifications"
                      class="pf-c-button pf-m-plain"
                      data-cy="notification_badge"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      title="Notifications"
                      type="button"
                    >
                      <span
                        class="pf-c-notification-badge pf-m-read"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                            transform=""
                          />
                        </svg>
                      </span>
                    </button>
                  </li>
                </ul>
                <button
                  aria-label="Scroll right"
                  class="pf-c-nav__scroll-button"
                  disabled=""
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      transform=""
                    />
                  </svg>
                </button>
              </nav>
            </div>
            <div
              class="pf-c-page__header-tools-item"
            >
              <div
                style="width: auto; height: 50px; justify-content: center; display: flex; align-items: center;"
              >
                <div
                  class="pf-c-dropdown"
                  data-ouia-component-id="1"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe="true"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="pf-c-dropdown__toggle pf-m-plain"
                    id="pf-toggle-id-0"
                    type="button"
                  >
                    <span
                      class="pf-c-dropdown__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                          transform=""
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="pf-c-page__header"
    >
      <div
        class="pf-c-page__header-brand"
      >
        <div
          class="pf-c-page__header-brand-toggle"
        >
          <button
            aria-controls="page-sidebar"
            aria-expanded="true"
            aria-label="Global navigation"
            class="pf-c-button pf-m-plain"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            id="nav-toggle"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 448 512"
              width="1em"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                transform=""
              />
            </svg>
          </button>
        </div>
        <a
          class="pf-c-page__header-brand-link"
        >
          <img
            alt="Open Innovation Labs"
            class="pf-c-brand"
            src="/oil_logo.png"
          />
        </a>
      </div>
      <div
        class="pf-c-page__header-tools"
      >
        <div
          class="pf-c-page__header-tools-group"
        >
          <div
            class="pf-c-page__header-tools-item"
          >
            <nav
              aria-label="Global"
              class="pf-c-nav pf-m-horizontal"
              data-ouia-component-id="2"
              data-ouia-component-type="PF4/Nav"
              data-ouia-safe="true"
            >
              <button
                aria-label="Scroll left"
                class="pf-c-nav__scroll-button"
                disabled=""
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                    transform=""
                  />
                </svg>
              </button>
              <ul
                class="pf-c-nav__list"
              >
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    data-cy="about_link"
                    href="/app/about"
                    style="padding-bottom: 0px; padding-top: 0px;"
                  >
                    <svg
                      aria-labelledby="icon-title-0"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 64 1024 1024"
                      width="1em"
                    >
                      <title
                        id="icon-title-0"
                      >
                        About
                      </title>
                      <path
                        d="M512.059-73.143c-282.338 0-512.059 229.673-512.059 512.025 0 282.235 229.721 511.975 512.059 511.975 282.281 0 511.941-229.735 511.941-511.975 0.005-282.352-229.659-512.025-511.941-512.025zM512.059 826.523c-213.826 0-387.728-173.856-387.728-387.643 0-213.89 173.904-387.694 387.728-387.694 213.717 0 387.671 173.803 387.671 387.694 0.005 213.785-173.957 387.643-387.671 387.643zM585.143 164.571v109.714c0 4.951-1.808 9.237-5.429 12.857s-7.906 5.429-12.857 5.429h-109.714c-4.953 0-9.239-1.808-12.857-5.429s-5.429-7.906-5.429-12.857v-109.714c0-4.951 1.81-9.237 5.429-12.857s7.904-5.429 12.857-5.429h109.714c4.951 0 9.237 1.808 12.857 5.429s5.429 7.906 5.429 12.857zM521.616 365.714c118.343 0 214.286 88.965 214.286 187.429s-95.943 178.286-214.286 178.286c-173.045 0-208.091-93.714-213.963-171.113 0-7.936 6.729-9.838 15.872-9.838s108.073 0 113.143 0c6.096 0 14.475 0.633 17.554 10.571 0 48.857 135.968 54.953 135.968-7.904 0-31.506-29.717-63.817-68.571-66.194s-82.286-7.607-82.286-54.199c0-13.022 0-25.673 0-44.693 0-19.015 9.717-22.343 27.431-22.343s54.853-0.002 54.853-0.002z"
                        transform="rotate(180 0 512) scale(-1 1)"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/app/requestfeature"
                    title="Feedback and Feature Request"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 352 512"
                      width="1em"
                    >
                      <path
                        d="M96.06 454.35c.01 6.29 1.87 12.45 5.36 17.69l17.09 25.69a31.99 31.99 0 0 0 26.64 14.28h61.71a31.99 31.99 0 0 0 26.64-14.28l17.09-25.69a31.989 31.989 0 0 0 5.36-17.69l.04-38.35H96.01l.05 38.35zM0 176c0 44.37 16.45 84.85 43.56 115.78 16.52 18.85 42.36 58.23 52.21 91.45.04.26.07.52.11.78h160.24c.04-.26.07-.51.11-.78 9.85-33.22 35.69-72.6 52.21-91.45C335.55 260.85 352 220.37 352 176 352 78.61 272.91-.3 175.45 0 73.44.31 0 82.97 0 176zm176-80c-44.11 0-80 35.89-80 80 0 8.84-7.16 16-16 16s-16-7.16-16-16c0-61.76 50.24-112 112-112 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
                        transform=""
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="pf-c-nav__item"
                >
                  <button
                    aria-label="Notifications"
                    class="pf-c-button pf-m-plain"
                    data-cy="notification_badge"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    title="Notifications"
                    type="button"
                  >
                    <span
                      class="pf-c-notification-badge pf-m-read"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
                          transform=""
                        />
                      </svg>
                    </span>
                  </button>
                </li>
              </ul>
              <button
                aria-label="Scroll right"
                class="pf-c-nav__scroll-button"
                disabled=""
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    transform=""
                  />
                </svg>
              </button>
            </nav>
          </div>
          <div
            class="pf-c-page__header-tools-item"
          >
            <div
              style="width: auto; height: 50px; justify-content: center; display: flex; align-items: center;"
            >
              <div
                class="pf-c-dropdown"
                data-ouia-component-id="1"
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="pf-c-dropdown__toggle pf-m-plain"
                  id="pf-toggle-id-0"
                  type="button"
                >
                  <span
                    class="pf-c-dropdown__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        transform=""
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
